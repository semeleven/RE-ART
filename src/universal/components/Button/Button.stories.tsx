import * as React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Button } from '../../components';

storiesOf('Button', module)
	.add('loading state', () => (
		<Button width="100%" loading onClick={action('click')}>
			BUTTON
		</Button>
	))
	.add('inverted dark color', () => (
		<Button invertedDark onClick={action('click')}>
			BUTTON
		</Button>
	))
	.add('inverted purple color', () => (
		<Button invertedPurple onClick={action('click')}>
			BUTTON
		</Button>
	))
	.add('dark color', () => (
		<Button dark onClick={action('click')}>
			BUTTON
		</Button>
	))
	.add('purple', () => (
		<Button purple onClick={action('click')}>
			BUTTON
		</Button>
	))
	.add('disabled', () => (
		<Button disabled onClick={action('click')}>
			BUTTON
		</Button>
	));
