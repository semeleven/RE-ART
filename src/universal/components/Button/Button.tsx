import React, { Component } from 'react';
import { FelaWithStylesProps } from 'react-fela';
import { darken } from 'polished';
import { withStyles, ThemeInterface } from '../../lib/fela/';

interface OwnProps {
	type: 'transparentDark' | 'transparentPurple' | 'purple'
	onClick: () => any
	theme: ThemeInterface
}

interface Styles {
	className: string
}

type Props = OwnProps & FelaWithStylesProps<OwnProps, Styles, ThemeInterface>

const styles = ({ 
	theme: { darkGray, darkPurple, white, purple },
    type 
}: Props): object => {
	const commonStyles = {
		boxSizing: 'border-box',
		fontFamily: 'Gotham',
		textTransform: 'uppercase',
		color: 'white',
		display: 'inline-flex',
		justifyContent: 'center',
		alignItems: 'center',
		minHeight: 40,
		minWidth: 115,
		paddingRight: 30,
		paddingLeft: 30,
		paddingTop: 11,
		paddingBottom: 11,
		borderStyle: 'solid',
		borderRadius: '10px',
		outline: 'none',
		fontSize: 15,
		textDecoration: 'none',
		verticalAlign: 'middle',
		cursor: 'pointer',
		transition: '0.1s ease-in-out',
	};

	if (type === 'transparentDark') {
		return {
			...commonStyles,
			borderColor: darkGray,
			color: darkGray,
			backgroundColor: white,
			'&:hover': {
				backgroundColor: darkGray,
				color: white,
			},
			'&:active': {
				backgroundColor: darken(0.1, darkGray),
			},
		}
	}

	if (type === 'transparentPurple') {
		return {
			...commonStyles,
			borderColor: purple,
			color: purple,
			backgroundColor: white,
			'&:hover': {
				backgroundColor: purple,
				color: white,
			},
			'&:active': {
				backgroundColor: darken(0.1, purple),
			},
		}
	}

	if (type === 'purple') {
		return {
			...commonStyles,
			borderColor: purple,
			color: white,
			backgroundColor: purple,
			'&:hover': {
				backgroundColor: darkPurple,
				color: white,
			},
			'&:active': {
				backgroundColor: darken(0.1, darkPurple),
			},
		}
	}
};

@withStyles(styles)
class Button extends Component<Props, any> {
	public render(): JSX.Element {
		const { className, children, onClick } = this.props;

		return (
			<button
				type="button"
				className={className}
				onClick={onClick}
			>
				{children}
			</button>
		);
	}
}

export default Button;

  