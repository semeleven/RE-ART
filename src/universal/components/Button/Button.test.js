import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import 'jest-styled-components';

import Button from './Button';

it('renders purple button', () => {
	const wrapper = shallow(<Button purple>PURPLE</Button>);
	expect(wrapper).toMatchSnapshot();
});

it('renders loading button', () => {
	const wrapper = shallow(
		<Button width="100%" loading onClick={() => {}}>
			SIGN UP
		</Button>
	);
	expect(wrapper).toMatchSnapshot();
});

it('simulates click events', () => {
	const onButtonClick = sinon.spy();
	const wrapper = shallow(<Button onClick={onButtonClick}>TEST</Button>);
	console.log(wrapper, 'wrapper?');
	wrapper.simulate('click');
	expect(onButtonClick.calledOnce).toEqual(true);
});
