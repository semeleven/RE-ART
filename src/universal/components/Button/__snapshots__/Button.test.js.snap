// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: default;
  width: 100%;
  background-color: #e2e2e2;
  color: #ffffff;
  border-style: solid;
  border-width: 2px;
  border-radius: 10px;
  box-sizing: border-box;
  font-family: 'Lato',sans-serif;
  font-weight: 600;
  text-transform: uppercase;
  text-align: center;
  padding: 11px 30px;
  outline: none;
  font-size: 15px;
  line-height: 1;
  cursor: pointer;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  background-color: #e2e2e2;
  border-color: #e2e2e2;
  color: white;
}

.c0:active {
  background-color: #e2e2e2;
  border-color: #e2e2e2;
  color: white;
}

<Button
  className="c0"
  loading={true}
  onClick={[Function]}
  width="100%"
>
  SIGN UP
</Button>
`;

exports[`renders purple button 1`] = `
.c0 {
  cursor: pointer;
  width: auto;
  background-color: #6C5FC7;
  color: #ffffff;
  border-color: #6C5FC7;
  border-style: solid;
  border-width: 2px;
  border-radius: 10px;
  box-sizing: border-box;
  font-family: 'Lato',sans-serif;
  font-weight: 600;
  text-transform: uppercase;
  text-align: center;
  padding: 11px 30px;
  outline: none;
  font-size: 15px;
  line-height: 1;
  cursor: pointer;
  -webkit-transition: 0.2s ease-in-out;
  transition: 0.2s ease-in-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c0:hover {
  background-color: #5346AE;
  border-color: #5346AE;
  color: white;
}

.c0:active {
  background-color: #42378a;
  border-color: #42378a;
  color: white;
}

<Button
  className="c0"
  purple={true}
>
  PURPLE
</Button>
`;
